datashuttle/configs.py

```diff
@@ -18,42 +18,102 @@ def __init__(self, file_path, input_dict):
         self.file_path = file_path
 
     def check_dict_values_and_inform_user(self):
-        """ """
-        if self["remote_path"][0] == "~":  # TODO: handle path or string
+        """
+        Check the values of the current dictionary are set
+        correctly and will not cause error (e.g. user has
+        set ssh_to_remote on but not set remote_path_ssh.
+        """
+
+        # Check relevant remote_path is set
+        if self["ssh_to_remote"]:
+            if not self["remote_path_ssh"]:
+                utils.raise_error(
+                    "ssh to remote is on but remote_path_ssh has not been set."
+                )
+        else:
+            if not self["remote_path_local"]:
+                utils.raise_error(
+                    "ssh to remote is off but remote_path_local has not been set."
+                )
+
+        # Check bad remote path format
+        if self.get_remote_path().as_posix()[0] == "~":
+        
             utils.raise_error(
                 "remote_path must contain the full directory path with no ~ syntax"
             )
 
+        # Check SSH settings
         if self["ssh_to_remote"] is True and (
             not self["remote_host_id"] or not self["remote_host_username"]
         ):
             utils.raise_error(
                 "ssh to remote set but no remote_host_id or remote_host_username not"
-                " provided"
+                " provided."
             )
 
         if self["ssh_to_remote"] is False and (
             self["remote_host_id"] or self["remote_host_username"]
         ):
             warnings.warn(
                 "SSH to remote is false, but remote_host_id or remote_host_username"
-                " provided"
+                " provided."
             )
 
     def update_an_entry(self, option_key: str, new_info: Union[str, bool]):
         """
         Convenience function to update individual entry of configuration file.
         The config file, and currently loaded self.cfg will be updated.
 
+        Note if user updates connection type, a different remote_path
```

Comment:
General point, but I would write these docstrings *for* the user, rather than for developers, referencing the user. If we release this with a Python API, we would expect them to dig into the docstrings.